### PR TITLE
MQTT Authz: Reduce test flakiness

### PR DESCRIPTION
**Pull request by vadim-kovalyov**
Our authorizers (EdgeHub and Policy) integraiton tests are flacky, because there were no way to deterministically wait until authorizer processes the update.

As a fix, instead of using time delay to wait for authorizer update (which is not deterministic way) , I have introduced a channel in `DummyAuthorizer` that signals when an update happens.

### CODE REVIEW

File: mqtt/mqtt-broker-tests-util/src/server.rs [hunk 1]
```diff
@@ -77,7 +77,7 @@ where
     P: MakePacketProcessor + Clone + Send + Sync + 'static,
 {
     lazy_static! {
-        static ref PORT: AtomicU32 = AtomicU32::new(5555);
+        static ref PORT: AtomicU32 = AtomicU32::new(8889);
```
Comment (vadim-kovalyov) [inline mqtt/mqtt-broker-tests-util/src/server.rs:80]: There is a chance of port collision if you have iotedge runtime running on your dev box. So I bumped it up to a more "safe" range of ports.

**Review by dmolokanov — APPROVED** [review]
Wow! Much better now! 🚀

File: mqtt/mqtt-edgehub/tests/common/mod.rs [hunk 2]
```diff
@@ -1,6 +1,7 @@
+#![allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (vadim-kovalyov) [reply mqtt/mqtt-edgehub/tests/common/mod.rs:1]: not really... for some reason in highlights DummyAuthorizer as not used, even though it's being used in tests.

**Review by and-rewsmith — APPROVED** [review]

File: mqtt/mqtt-edgehub/tests/common/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmolokanov) [reply mqtt/mqtt-edgehub/tests/common/mod.rs:1]: let's move it to `DummyAuthorizer` level to reduce the scope then?

**Review by dmolokanov — APPROVED** [review]

**Pull request merged**